lib/phoenix/controller.ex

```diff
@@ -468,26 +472,40 @@ defmodule Phoenix.Controller do
   `layout_formats/2` and `put_layout_formats/2` can be used to configure
   which formats support/require layout rendering (defaults to "html" only).
   """
-  @spec render(Plug.Conn.t, binary | atom, Dict.t) :: Plug.Conn.t
-  def render(conn, template, assigns) when is_atom(template) do
+  @spec render(Plug.Conn.t, binary | atom, atom | binary | Dict.t) :: Plug.Conn.t
+  def render(conn, template, assigns)
+    when is_atom(template) and is_list(assigns) do
     format =
       conn.params["format"] ||
       raise "cannot render template #{inspect template} because conn.params[\"format\"] is not set. " <>
             "Please set `plug :accepts, %w(html json ...)` in your pipeline."
-    render(conn, template_name(template, format), format, assigns)
+    do_render(conn, template_name(template, format), format, assigns)
   end
 
   def render(conn, template, assigns) when is_binary(template) do
     case Path.extname(template) do
       "." <> format ->
-        render(conn, template, format, assigns)
+        do_render(conn, template, format, assigns)
       "" ->
         raise "cannot render template #{inspect template} without format. Use an atom if the " <>
               "template format is meant to be set dynamically based on the request format"
     end
   end
 
-  def render(conn, template, format, assigns) do
+  def render(conn, view, template)
+    when is_atom(view) and is_binary(template) or is_atom(template) do
+    render(conn, view, template, [])
+  end
+
+  @spec render(Plug.Conn.t, atom, atom | binary, Dict.t) :: Plug.Conn.t
+  def render(conn, view, template, assigns)
```

Comment:
We should rather make this call the canonical one and have the other variants calling here so we avoid the put_view indirection!